### PR TITLE
Increase epsilon in SE(2/3) Tangent Computation (fixes #42)

### LINKED ISSUE #42: Some weird behavior when using small input u
I'm using the code in [test](https://github.com/tud-phi/soromox/blob/main/examples/simulation/pcs/simulate_tendon_actuated_pcs.py) for TendonActuatedPCS. The parameters I'm using is (which is almost like the example):

      num_segments = 1
      rho = 1070 * jnp.ones(
          (num_segments,)
      )  # Volumetric density of Dragon Skin 20 [kg/m^3]
      params = {
          "p0": jnp.array([jnp.pi / 2, jnp.pi / 2, 0.0, 0.0, 0.0, 0.0]),  # Initial position and orientation
          "L": 0.25 * jnp.ones((num_segments,)),  # default: 1e-1
          "r": 0.025 * jnp.ones((num_segments,)),  # default: 2e-2
          "rho": rho,
          "g": jnp.array([0.0, 0.0, 0.0]),  # Gravity vector [m/s^2]
          "E": 2.5e5 * jnp.ones((num_segments,)),  # Elastic modulus [Pa]
          "G": 0.912e5 * jnp.ones((num_segments,)),  # Shear modulus [Pa]
      }
      params["D"] = 1e-3 * jnp.diag(
          (
              jnp.repeat(
                  jnp.array([[1e0, 1e0, 1e0, 1e3, 1e3, 1e3]]), num_segments, axis=0
              )
              * params["L"][:, None]
          ).flatten()
      )
      active_tendon_routing_params = {
              "ry": jnp.array([0.02, -0.02, 0.0, 0.0]),
              "rz": jnp.array([0.0, 0.0, 0.02, -0.02]),
              "my": jnp.array([0.0, 0.0, 0.0, 0.0]),
              "mz": jnp.array([0.0, 0.0, 0.0, 0.0]),
              "idx_seg_att": jnp.array([0, 0, 0, 0]),    
          }
  

    # ======================================================
    # Robot initialization
    # ======================================================

    robot = TendonActuatedPCS(
        num_segments=num_segments,
        params=params,
        active_tendon_routing_params=active_tendon_routing_params,
    )

    # =====================================================
    # Simulation upon time
    # =====================================================
    # Initial configuration

    q0 = jnp.repeat(
        jnp.array([0.0, 0.0,0.0,0.0,0.0, 0.0])[None, :],
        num_segments,
        axis=0,
    ).flatten()
    # Initial velocities
    qd0 = jnp.zeros_like(q0)

And I found with different input u, it shows different behavior but weird:

With u=[1.0, 0.0, 1.0, 0.0], the behavior is like:

https://github.com/user-attachments/assets/13112a96-60ee-49e5-9b82-56a166670179

With u=[0.05, 0.0, 0.05, 0.0], the behavior is like: (I think it is not correct)

https://github.com/user-attachments/assets/159e8255-a6be-435e-8a2f-163869a26f70

With u = [0.005, 0.0, 0.005, 0.0], the behavior is like: (I think it is good now)

https://github.com/user-attachments/assets/da87b9b7-4d85-498f-8007-1f2a36139704

Could I ask why it happens? Thank you very much!

### PR DESCRIPTION
**Pull request by mstoelzle**
This pull request refactors how the epsilon parameter used for numerical stability is handled across all dynamical system classes. Instead of each subclass defining its own `global_eps`, the epsilon is now defined once in the base `DynamicalSystem` class and accessed via a new property, ensuring consistency and reducing redundancy. 
Even more importantly, we are defining a separate `tangent_eps` to be used for SE(2) & SE(3) tangent and tangent derivative computations.
All relevant method calls are updated to use this new approach. Additionally, type annotations are improved for clarity in utility functions.

**Refactoring of epsilon handling across dynamical systems:**

* Introduced `global_eps` as a `ClassVar` and a `tangent_eps` property in `DynamicalSystem`, centralizing the definition of epsilon for all subclasses. (`src/soromox/systems/dynamical_system.py`) [[1]](diffhunk://#diff-01e0052383ed2a1884d67c784069cc19a09b5253ab5ae23fd9e3ed1556bc4522L7-R7) [[2]](diffhunk://#diff-01e0052383ed2a1884d67c784069cc19a09b5253ab5ae23fd9e3ed1556bc4522R26-R33)
* Removed redundant `global_eps` definitions from subclasses `GVS`, `PCS`, `PlanarHSA`, and `PlanarPCS`. (`src/soromox/systems/gvs/core.py`, `src/soromox/systems/pcs.py`, `src/soromox/systems/planar_hsa.py`, `src/soromox/systems/planar_pcs.py`) [[1]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L137-L138) [[2]](diffhunk://#diff-19b0b4a682b4e94d1538d4dd92005bf5c097e7cabb76cb4ddf8c61e55fc851c0L87-L89) [[3]](diffhunk://#diff-b5d02dd98c17946883b02ddb989960fb0b0a280384b0240e91a9c46667e6232cL204-L206) [[4]](diffhunk://#diff-4397bec6ccb864d3bfb2d4a8e482b3312663630e95789bf9d8ada63ffae30b5cL84-L86)

**Consistent usage of epsilon in Lie algebra computations:**

* Updated all calls to `lie.Tangent_gi_se3`, `lie.Tangent_derivative_gi_se3`, `lie.Tangent_gi_se2`, and `lie.Tangent_derivative_gi_se2` to use `self.tangent_eps` instead of `self.global_eps`, ensuring consistent numerical stability across all computations. [[1]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L1218-R1216) [[2]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L1291-R1289) [[3]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L1403-R1401) [[4]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L1469-R1467) [[5]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L1554-R1552) [[6]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L1658-R1659) [[7]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L1744-R1745) [[8]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L1872-R1872) [[9]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L1956-R1956) [[10]](diffhunk://#diff-a09d10506394df2d23096266f1509bdec92074df3b65578f7dddc7aa619083a6L2056-R2056) [[11]](diffhunk://#diff-19b0b4a682b4e94d1538d4dd92005bf5c097e7cabb76cb4ddf8c61e55fc851c0L785-R782) [[12]](diffhunk://#diff-19b0b4a682b4e94d1538d4dd92005bf5c097e7cabb76cb4ddf8c61e55fc851c0L852-R849) [[13]](diffhunk://#diff-19b0b4a682b4e94d1538d4dd92005bf5c097e7cabb76cb4ddf8c61e55fc851c0L910-R907) [[14]](diffhunk://#diff-19b0b4a682b4e94d1538d4dd92005bf5c097e7cabb76cb4ddf8c61e55fc851c0L999-R998) [[15]](diffhunk://#diff-19b0b4a682b4e94d1538d4dd92005bf5c097e7cabb76cb4ddf8c61e55fc851c0L1100-R1101) [[16]](diffhunk://#diff-19b0b4a682b4e94d1538d4dd92005bf5c097e7cabb76cb4ddf8c61e55fc851c0L1207-R1206) Fc11705aL896R894, [[17]](diffhunk://#diff-4397bec6ccb864d3bfb2d4a8e482b3312663630e95789bf9d8ada63ffae30b5cL1000-R997) [[18]](diffhunk://#diff-4397bec6ccb864d3bfb2d4a8e482b3312663630e95789bf9d8ada63ffae30b5cL1048-R1045) [[19]](diffhunk://#diff-4397bec6ccb864d3bfb2d4a8e482b3312663630e95789bf9d8ada63ffae30b5cL1149-R1148) [[20]](diffhunk://#diff-4397bec6ccb864d3bfb2d4a8e482b3312663630e95789bf9d8ada63ffae30b5cL1289-R1290) [[21]](diffhunk://#diff-4397bec6ccb864d3bfb2d4a8e482b3312663630e95789bf9d8ada63ffae30b5cL1380-R1379)

**Type annotation improvements in SE(2) utilities:**

* Updated the type annotation for the `eps` parameter in `log_SE2` to accept both `float` and `Array`, improving flexibility and clarity. (`src/soromox/utils/lie_algebra/se2.py`) [[1]](diffhunk://#diff-7a135597bd290ab5b0ff71144206228e72acd094c018c9c69095527bcb3939a0R20) [[2]](diffhunk://#diff-7a135597bd290ab5b0ff71144206228e72acd094c018c9c69095527bcb3939a0L90-R96)

### CODE REVIEW

**Comment by Hhannzch** [comment]
Hi Maxi,

I found another error case from training: when u = [0.1119 0.     0.     0.    ] or u = [0.12 0.     0.     0.    ].🤔

**Review by Hhannzch — APPROVED** [review]
Thanks Maxi 👍
